spec.html

```diff
@@ -7200,46 +7220,41 @@ <h1>CreateImportBinding ( _N_, _M_, _N2_ )</h1>
         <emu-alg>
           1. If _lex_ is the value *null*, then
             1. Return a value of type Reference whose base value component is *undefined*, whose referenced name component is _name_, and whose strict reference flag is _strict_.
-          1. Let _envRec_ be _lex_'s EnvironmentRecord.
-          1. Let _exists_ be ? _envRec_.HasBinding(_name_).
+          1. Let _exists_ be ? _lex_.HasBinding(_name_).
           1. If _exists_ is *true*, then
-            1. Return a value of type Reference whose base value component is _envRec_, whose referenced name component is _name_, and whose strict reference flag is _strict_.
+            1. Return a value of type Reference whose base value component is _lex_, whose referenced name component is _name_, and whose strict reference flag is _strict_.
           1. Else,
-            1. Let _outer_ be the value of _lex_'s outer environment reference.
+            1. Let _outer_ be _lex_.[[OuterEnv]].
             1. Return ? GetIdentifierReference(_outer_, _name_, _strict_).
         </emu-alg>
       </emu-clause>
 
       <emu-clause id="sec-newdeclarativeenvironment" aoid="NewDeclarativeEnvironment">
         <h1>NewDeclarativeEnvironment ( _E_ )</h1>
-        <p>The abstract operation NewDeclarativeEnvironment takes argument _E_ (a Lexical Environment). It performs the following steps when called:</p>
+        <p>The abstract operation NewDeclarativeEnvironment takes argument _E_ (an Environment Record). It performs the following steps when called:</p>
         <emu-alg>
-          1. Let _env_ be a new Lexical Environment.
-          1. Let _envRec_ be a new declarative Environment Record containing no bindings.
-          1. Set _env_'s EnvironmentRecord to _envRec_.
-          1. Set the outer lexical environment reference of _env_ to _E_.
+          1. Let _env_ be a new declarative Environment Record containing no bindings.
+          1. Set _env_.[[OuterEnv]] to _E_.
           1. Return _env_.
         </emu-alg>
       </emu-clause>
 
       <emu-clause id="sec-newobjectenvironment" aoid="NewObjectEnvironment">
         <h1>NewObjectEnvironment ( _O_, _E_ )</h1>
-        <p>The abstract operation NewObjectEnvironment takes arguments _O_ (an Object) and _E_ (a Lexical Environment). It performs the following steps when called:</p>
+        <p>The abstract operation NewObjectEnvironment takes arguments _O_ (an Object) and _E_ (an Environment Record). It performs the following steps when called:</p>
         <emu-alg>
-          1. Let _env_ be a new Lexical Environment.
-          1. Let _envRec_ be a new object Environment Record containing _O_ as the binding object.
-          1. Set _env_'s EnvironmentRecord to _envRec_.
-          1. Set the outer lexical environment reference of _env_ to _E_.
+          1. Let _env_ be a new object Environment Record containing _O_ as the binding object.
```

Comment:
I would prefer the binding object be turned into a slot on the object Environment Record. We can save that for another PR, though.

Reply:
Agreed on both points. (Adding it to my to-do.)